salt/modules/composer.py

```diff
@@ -185,9 +189,14 @@ def _run_composer(action,
     if optimize is True:
         cmd.append('--optimize-autoloader')
 
+    if env is not None:
+        env.update({'COMPOSER_HOME': composer_home})
```

Comment:
This assumes that the env vars are in a YAML dict:

```yaml
mystate:
  statemod.funcname:
    - name: foo
    - env:
        FOO: bar
        HELLO: world
```

But because of some weirdness in PyYAML which requires nested dicts to be double-indented to be properly loaded, in many places in Salt we also allow input to be a bit more user-friendly, like so:

```yaml
mystate:
  statemod.funcname:
    - name: foo
    - env:
      - FOO: bar
      - HELLO: world
```

We have a helper function, `salt.utils.data.repack_dictlist()` which you can use here:

```python
if env is not None:
    env = salt.utils.data.repack_dictlist(env)
    env['COMPOSER_HOME'] = composer_home
```

Reply:
That helper will load the YAML list of dicts into a single dict.

Reply:
Thank you for that information @terminalmage 
I was actually having some difficulties with exactly that 👍